eva-server/src/test/java/uk/ac/ebi/eva/server/ws/FilesWSServerTest.java

```diff
@@ -18,64 +18,89 @@
+                VariantStudy.StudyType.COLLECTION, VariantSource.Aggregation.NONE, samples, metadata,
+                variantGlobalStats);
+        List<VariantSourceEntity> variantSourceEntities = Collections.singletonList(variantSourceEntity);
+
+        given(variantSourceEntityRepository.findAll()).willReturn(variantSourceEntities);
     }
     
     @Test
     public void testGetFiles() throws URISyntaxException {
-        Response response = given().param("species", "mmusculus_grcm38").get(new URI("/v1/files/all"));
-        response.then().statusCode(200);
-        
-        List queryResponse = JsonPath.from(response.asString()).getList("response");
-        assertEquals(1, queryResponse.size());
-        
-        List<Map> result = JsonPath.from(response.asString()).getJsonObject("response[0].result");
-        assertTrue(result.size() >= 1);
-        
-        for (int i = 0; i < result.size() ; i++) {
-            Map m = result.get(i);
-            String missingField = String.format("%s required field missing", m.get("name"));
-            
-            assertTrue(missingField, m.containsKey("fileName"));
-            assertTrue(missingField, m.containsKey("fileId"));
-            assertTrue(missingField, m.containsKey("studyName"));
-            assertTrue(missingField, m.containsKey("studyId"));
-            assertTrue(missingField, m.containsKey("aggregation"));
-            assertTrue(missingField, m.containsKey("type"));
-            assertTrue(missingField, m.containsKey("metadata"));
-            assertTrue(missingField, m.containsKey("samplesPosition"));
-            
-            if (m.containsKey("stats")) {
-                Map<String, ?> stats = JsonPath.from(response.asString()).getJsonObject("response[0].result[" + i + "].stats");
-                assertTrue(missingField, stats.containsKey("variantsCount"));
-                assertTrue(missingField, stats.containsKey("samplesCount"));
-                assertTrue(missingField, stats.containsKey("snpsCount"));
-                assertTrue(missingField, stats.containsKey("indelsCount"));
-                assertTrue(missingField, stats.containsKey("structuralCount"));
-                assertTrue(missingField, stats.containsKey("passCount"));
-                assertTrue(missingField, stats.containsKey("transitionsCount"));
-                assertTrue(missingField, stats.containsKey("transversionsCount"));
-                assertTrue(missingField, stats.containsKey("accumulatedQuality"));
-                assertTrue(missingField, stats.containsKey("meanQuality"));
-            }
+        String url = "/v1/files/all?species=hsapiens_grch37";
+        ResponseEntity<QueryResponse<QueryResult<VariantSourceEntity>>> response = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<VariantSourceEntity>>>() {});
```

Comment:
You are right, this is the most type-safe solution if the entities are not returned straight away.